### PR TITLE
Fix #3788: Break `*Required` into two parts.

### LINKED ISSUE #3788: Break AddProfileActivity string "*Required" into two parts.
`“*Required”` string in `Add Profile` should be broken down into two parts because “*” does not need a translation.

### PR DESCRIPTION
**Pull request by rishidyno**
Fix #3788
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by rishidyno** [comment]
@rt4914 I have a doubt 
when I get the string resource for *Required 
`val starRequired: String = Resources.getSystem().getString(R.string.add_profile_required)`
the app keeps crashing when i open add_profie_activity but not in the case of 
`val starRequired: String = "*Required"`

File: app/src/main/java/org/oppia/android/app/profile/AddProfileViewModel.kt [hunk 1]
```diff
@@ -18,6 +18,7 @@ class AddProfileViewModel @Inject constructor() : ObservableViewModel() {
   val createPin = ObservableField(false)
   val isButtonActive = ObservableField(false)
   val showInfoAlertPopup = ObservableField<Boolean>(false)
+  val starRequired: String = "*Required"
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/profile/AddProfileViewModel.kt]: Here `*` should be mentioned here by `required` should come from `strings.xml` file.

Also whatever error you are getting can you mention it here or create a [debugging doc](https://github.com/oppia/oppia/wiki/Debugging-Docs) mentioning what steps you performed and what errors you got.

File: app/src/sharedTest/java/org/oppia/android/app/profile/AddProfileActivityTest.kt [hunk 2]
```diff
@@ -172,6 +173,15 @@ class AddProfileActivityTest {
     assertThat(label).isEqualTo(context.getString(R.string.add_profile_activity_label))
   }
 
+  @Test
+  fun testAddProfileActivity_hasCorrectRequiredString() {
+    activityTestRule.launchActivity(createAddProfileActivityIntent())
+    val label = activityTestRule.activity.add_profile_activity_required_heading_text_view.text
+
+    // Verify that the activity shows the correct *Required string for add_profile_activity_required_heading_text_view
+    assertThat(label).isEqualTo(context.getString(R.string.add_profile_required))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/profile/AddProfileActivityTest.kt]: This test case has been written incorrectly. Please check below test case in this file `testAddProfileActivity_inputName_opensProfileChooserActivity ` and implement a test similar to that.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@rishidyno PTAL

File: app/src/main/java/org/oppia/android/app/profile/AddProfileViewModel.kt (hunk 1 above)
Comment (rishidyno) [reply app/src/main/java/org/oppia/android/app/profile/AddProfileViewModel.kt]: I did not get this properly could you please make it clear to me again.

Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/profile/AddProfileViewModel.kt]: Part 1 of string = *
Part 2 of string = Required.

Part 2 should be mentioned in `strings.xml` file. and Part 1 should be directly inside kotlin.
@rishidyno If this does not make sense ping me and we can meet.

**Comment by rt4914** [comment]
@rishidyno Any updates?

**Comment by rishidyno** [comment]
> @rishidyno Any updates?

@rt4914  Sorry! I was away because of my personal work. I will continue now.

**Comment by rishidyno** [comment]
> @rishidyno Any updates?

@rt4914 I am actually out of station for work. I will continue on 20th

**Comment by rishidyno** [comment]
Screenshot from 2021-10-22 12-24-10


@rt4914 here I am getting an unresolved reference error could you please help me out with this.
 Also, please give a verdict on the test I have written.

**Pull request closed without merging**